core/chaincode/lifecycle/lifecycle.go

```diff
@@ -407,6 +407,24 @@ func (ef *ExternalFunctions) ApproveChaincodeDefinitionForOrg(chname, ccname str
 	}
 
 	privateName := fmt.Sprintf("%s#%d", ccname, requestedSequence)
+	uncommittedMetadata, ok, err := ef.Resources.Serializer.DeserializeMetadata(NamespacesName, privateName, orgState)
+	if err != nil {
+		return errors.WithMessage(err, "could not fetch uncommitted definition")
+	}
+
+	if ok {
+		logger.Debugf("Attempting to redefine uncommitted definition at sequence %d", requestedSequence)
+
+		uncommittedParameters := &ChaincodeParameters{}
+		if err := ef.Resources.Serializer.Deserialize(NamespacesName, privateName, uncommittedMetadata, uncommittedParameters, orgState); err != nil {
+			return errors.WithMessagef(err, "could not deserialize namespace %s as chaincode", privateName)
+		}
+
+		if err := uncommittedParameters.Equal(cd.Parameters()); err == nil {
+			return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
```

Comment:
This breaks a (bad?) test:
https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=5076&view=logs&j=17737827-9315-5d9e-c14a-16a17e90ef50&t=4c34d36c-6bff-5a33-cfcd-b209c5bac4c3&l=4342

Reply:
test is correct. i forgot the case that pkg id can still be updated after def being committed. fixed the code (should've wait for CI go green before submitting for review...)